medium-tests/src/test/java/mediumtest/hotspots/HotspotEventsMediumTests.java

```diff
@@ -265,6 +267,90 @@ void it_should_update_hotspot_in_storage_when_changing_status(SonarLintTestHarne
         .containsOnly(tuple("AYhSN6mVrRF_krvNbHl1", HotspotReviewStatus.SAFE)));
     }
 
+    @SonarLintTest
+    void it_should_update_known_findings_store_when_changing_status(SonarLintTestHarness harness, @TempDir Path baseDir) {
+      // create a file with hotspot in it
+      var filePath = createFile(baseDir, "Foo.java", """
+        public class Foo {
+          String ip = "192.168.12.42"; // Sensitive
+          Socket socket = new Socket(ip, 6667);
+        }
+        """);
+      var fileUri = filePath.toUri();
+
+      var connectionId = "connectionId";
+      var branchName = "branchName";
+      String projectKey = "projectKey";
+      var serverHotspotKey = "myHotspotKey";
+
+      var client = harness.newFakeClient()
+        .withToken(connectionId, "token")
+        .withInitialFs(CONFIG_SCOPE_ID, baseDir, List.of(new ClientFileDto(fileUri, baseDir.relativize(filePath), CONFIG_SCOPE_ID, false, null, filePath, null, null, true)))
+        .withMatchedBranch(CONFIG_SCOPE_ID, branchName)
+        .build();
+
+      var server = harness.newFakeSonarQubeServer("10.4")
+        .withProject(projectKey,
+          project -> project
+            .withBranch(branchName))
+        .withServerSentEventsEnabled()
+        .start();
+
+      // initialize backend with pre-filled storage with matching hotspot and active rule
+      var backend = harness.newBackend()
+        .withExtraEnabledLanguagesInConnectedMode(JAVA)
+        .withServerSentEventsEnabled()
+        .withSecurityHotspotsEnabled()
+        .withSonarQubeConnection(connectionId, server,
+          storage -> storage.withProject(projectKey,
+            project -> project
+              .withMainBranch(branchName, branch -> branch.withHotspot(aServerHotspot(serverHotspotKey)
+                .withStatus(HotspotReviewStatus.TO_REVIEW)
+                .withFilePath(baseDir.relativize(filePath).toString())
+                .withTextRange(new TextRangeWithHash(2, 14, 2, 29, "c50a46d24d0975188e037e408583ad30"))
+                .withRuleKey("java:S1313")
+                .withMessage("Make sure using this hardcoded IP address is safe here.")
+                .withVulnerabilityProbability(VulnerabilityProbability.LOW)
+              ))
+              .withRuleSet("java", ruleSetBuilder -> ruleSetBuilder.withActiveRule("java:S1313", "BLOCKER"))
+              )
```

Comment:
The indentation seems a bit off here, are you using the right formatting settings?